pkg/domain/infra/abi/quadlet.go

```diff
@@ -0,0 +1,764 @@
+// It handles both individual Quadlet files and Quadlet applications (collections of related files).
+// The function performs the following operations:
+//   - Validates that specified Quadlet files exist
+//   - Checks if associated systemd services are running
+//   - Optionally stops running services (if force option is set)
+//   - Removes the Quadlet files and any associated asset files
+//   - Reloads systemd daemon if necessary
+//
+// Parameters:
+//   - ctx: Context for cancellation and timeout handling
+//   - quadlets: List of Quadlet names to remove, or empty slice if options.All is true
+//   - options: Configuration options including Force, Ignore, and All flags
+//
+// Returns:
+//   - QuadletRemoveReport: Contains lists of successfully removed quadlets and any errors encountered
+//   - error: Non-nil if a fatal error occurred that prevented the operation from completing
+//
+// The function supports:
+//   - Force removal of running Quadlet services
+//   - Ignoring missing Quadlet files (with Ignore option)
+//   - Removing all installed Quadlets (with All option)
+//   - Handling Quadlet applications (.app files containing multiple related Quadlets)
+func (ic *ContainerEngine) QuadletRemove(ctx context.Context, quadlets []string, options entities.QuadletRemoveOptions) (*entities.QuadletRemoveReport, error) {
+	report := entities.QuadletRemoveReport{
+		Errors:  make(map[string]error),
+		Removed: []string{},
+	}
+	removeList := []string{}
+	reverseMap, appMap, err := buildAppMap(systemdquadlet.GetInstallUnitDirPath(rootless.IsRootless()))
+	if err != nil {
+		return nil, fmt.Errorf("unable to build app map: %w", err)
+	}
+	quadletsRe := []string{}
+	// Process all `.app` files in arguments, if `.app` file
+	// is found then expand it to its respective quadlet files
+	// and remove it from the processing list.
+	for _, quadlet := range quadlets {
+		// Most likely this is an app
+		if strings.HasPrefix(quadlet, ".") && strings.HasSuffix(quadlet, ".app") {
+			files, ok := appMap[quadlet]
+			// Add all files of this application in to-be removed list.
+			if ok {
+				for _, file := range files {
+					if !systemdquadlet.IsExtSupported(file) {
+						removeList = append(removeList, file)
+					} else {
+						quadletsRe = append(quadletsRe, file)
+					}
+				}
+			}
+			// also add .app file itself to the remove list so it can
+			// be cleaned after removal of all components in the list
+			if !slices.Contains(removeList, quadlet) {
+				removeList = append(removeList, quadlet)
+			}
+		} else {
+			quadletsRe = append(quadletsRe, quadlet)
+		}
+	}
+	quadlets = quadletsRe
+	allQuadletPaths := make([]string, 0, len(quadlets))
+	allServiceNames := make([]string, 0, len(quadlets))
+	runningQuadlets := make([]string, 0, len(quadlets))
+	serviceNameToQuadletName := make(map[string]string)
+	needReload := false
+
+	// Early escape: if 0 quadlets are requested, bail immediately without error.
```

Comment:
The comment is just straight up wrong. We error out not without error. And I mean the comment text itself doesn't add value because it is clear from the code what it does. code comments need to say why not what unless the what itself is hard to follow/understand.